packages/jest/src/automatic.ts

```diff
@@ -0,0 +1,58 @@
+/*
+ * Copyright (c) 2021, salesforce.com, inc.
+ * All rights reserved.
+ * SPDX-License-Identifier: BSD-3-Clause
+ * For full license text, see the LICENSE file in the repo root or https://opensource.org/licenses/BSD-3-Clause
+ */
+
+/**
+ * Options for Automatic checks to be passed to {@link registerSa11yAutomaticChecks}
+ */
+export type AutoCheckOpts = {
+    runAfterEach?: boolean;
+    cleanupAfterEach?: boolean;
+    // TODO (feat): add support for result consolidation across all tests for a test run
+    //  and optional exclusion of selected tests
+    // consolidateResults?: boolean;
+    // excludeTests?: string[];
+};
+
+/**
+ * Default options when {@link registerSa11yAutomaticChecks} is invoked
+ */
+const defaultAutoCheckOpts: AutoCheckOpts = {
+    runAfterEach: true,
+    cleanupAfterEach: true,
+};
+
+/**
+ * Run accessibility check on each element node in the body using {@link toBeAccessible}
+ * @param cleanup - boolean indicating if the DOM should be cleaned up after
+ */
+export async function automaticCheck(cleanup = false): Promise<void> {
+    let currNode = document.body.firstElementChild;
+    while (currNode !== null) {
+        try {
+            // TODO (spike): Use a logger lib with log levels selectable at runtime
+            // console.log('♿ [DEBUG] Automatically checking a11y of ' + currNode.nodeName);
+            await expect(currNode).toBeAccessible();
+        } finally {
+            if (cleanup) document.body.removeChild(currNode);
+            currNode = currNode.nextElementSibling;
```

Comment:
From my local testing, `currNode` will be null here after the first run since you already removed it from the DOM in the line above. 

Also, the TreeWalker API does have a `nextSibling` API - https://developer.mozilla.org/en-US/docs/Web/API/TreeWalker/nextSibling

Reply:
Ah 🤦 thanks for the catch - had some suspicion around that logic - should have checked it when I was debugging the flow to fix the earlier bug in browser console instead of misreading the docs that it would continue to work in detached state. 

Had been wondering from the bug discovered earlier that the test data was insufficient - because it had only 1 node with a11y issue. If there were multiple child nodes with a11y issues it might have helped catch this issue too. Fixed that now by adding another child node with a11y issue to the test data.

Switched back to using tree walker logic. Yes noticed earlier that treewalker had `nextSibling` but was looking for a way to consolidate the `firstChild` outside the loop and `nextSibling` inside the loop with something like `while (walker.nextChild()) ...`

Also did some refactoring to consolidate errors from all elements in the body instead of just the first one found.